internal/controllers/mbsc_reconciler.go

```diff
@@ -0,0 +1,196 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	kmmv1beta1 "github.com/kubernetes-sigs/kernel-module-management/api/v1beta1"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/api"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/buildsign"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/kernel"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/mbsc"
+	"github.com/kubernetes-sigs/kernel-module-management/internal/utils"
+	v1 "k8s.io/api/core/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const MBSCReconcilerName = "MBSCReconciler"
+
+// mbscReconciler reconciles a ModuleBuldSignConfig object
+type mbscReconciler struct {
+	reconHelperAPI mbscReconcilerHelperAPI
+}
+
+func NewMBSCReconciler(
+	client client.Client,
+	buildSignAPI buildsign.Manager,
+	mbscAPI mbsc.MBSC,
+) *mbscReconciler {
+	reconHelperAPI := newMBSCReconcilerHelper(client, buildSignAPI, mbscAPI)
+	return &mbscReconciler{
+		reconHelperAPI: reconHelperAPI,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *mbscReconciler) SetupWithManager(mgr ctrl.Manager, kernelLabel string) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&kmmv1beta1.ModuleBuildSignConfig{}).
+		Owns(&v1.Pod{}).
+		Named(MBSCReconcilerName).
+		Complete(
+			reconcile.AsReconciler[*kmmv1beta1.ModuleBuildSignConfig](mgr.GetClient(), r),
+		)
+}
+
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modulebuildsignconfigs,verbs=get;list;watch;update;patch;create
+//+kubebuilder:rbac:groups=kmm.sigs.x-k8s.io,resources=modulebuildsignconfigs/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups="core",resources=secrets,verbs=get;list;watch
+//+kubebuilder:rbac:groups="core",resources=configmaps,verbs=get;list;watch
```

Comment:
Can we combine those 2 resources since they have the same verbs?

Reply:
I would prefer to keep them separate, for readability sake. It already happened to me that i missed the resource, because it was in the same line as another resource